src/chart/generateCategoricalChart.tsx

```diff
@@ -2184,6 +2185,18 @@ export const generateCategoricalChart = ({
             }),
           ];
         }
+      } else if (hasActive) {
+        /**
+         * We hit this block if consumer uses a Tooltip without XAxis and/or YAxis.
+         * In which case, this.state.activeTooltipIndex never gets set
+         * because the mouse events that trigger that value getting set never get trigged without the axis components.
+         *
+         * See FunnelChart story for an example use case
+         */
+        const { data: itemData = [] } = graphicalItem.props;
+        const activeData = itemData.find((datum: { value: number }) => datum.value === activeItem.payload.value);
+        const activeIndex = itemData.indexOf(activeData);
```

Comment:
How about duplicates in the data? 🤔

Reply:
Nice catch. I've refactored how I'm calculating the active index, and added a duplicate in the data set for FunnelChart story. Please give my changes a look.